src/query/src/part_sort.rs

```diff
@@ -0,0 +1,545 @@
+                .into_iter()
+                .map(|(range, data)| {
+                    let part = PartitionRange {
+                        start: Timestamp::new(range.0, unit.clone().into()),
+                        end: Timestamp::new(range.1, unit.clone().into()),
+                        num_rows: data.iter().map(|b| b.len()).sum(),
+                        identifier,
+                    };
+
+                    let batches = data
+                        .into_iter()
+                        .map(|b| {
+                            let arr = new_ts_array(unit.clone(), b);
+                            DfRecordBatch::try_new(schema.clone(), vec![arr]).unwrap()
+                        })
+                        .collect_vec();
+                    (part, batches)
+                })
+                .collect_vec();
+
+            let expected_output = expected_output
+                .into_iter()
+                .map(|a| {
+                    DfRecordBatch::try_new(schema.clone(), vec![new_ts_array(unit.clone(), a)])
+                        .unwrap()
+                })
+                .collect_vec();
+
+            run_test(0, input_ranged_data, schema.clone(), opt, expected_output).await;
+        }
+    }
+
+    async fn run_test(
+        case_id: usize,
+        input_ranged_data: Vec<(PartitionRange, Vec<DfRecordBatch>)>,
+        schema: SchemaRef,
+        opt: SortOptions,
+        expected_output: Vec<DfRecordBatch>,
+    ) {
+        let (_ranges, batches): (Vec<_>, Vec<_>) = input_ranged_data.clone().into_iter().unzip();
+
+        let batches = batches
+            .into_iter()
+            .flat_map(|mut cols| {
+                cols.push(DfRecordBatch::new_empty(schema.clone()));
+                cols
+            })
+            .collect_vec();
+        let mock_input = MockInputExec::new(batches, schema.clone());
+
+        let exec = PartSortExec::try_new(
+            PhysicalSortExpr {
+                expr: Arc::new(Column::new("ts", 0)),
+                options: opt,
+            },
+            Arc::new(mock_input),
+        )
+        .unwrap();
+
+        let exec_stream = exec.execute(0, Arc::new(TaskContext::default())).unwrap();
+
+        let real_output = exec_stream.map(|r| r.unwrap()).collect::<Vec<_>>().await;
+
+        // a makeshift solution for compare large data
+        if real_output != expected_output {
+            {
+                let mut f_res = std::io::stderr();
+                let mut buf = String::new();
+                for batch in &real_output {
+                    let mut rb_json: Vec<u8> = Vec::new();
+                    let mut writer = ArrayWriter::new(&mut rb_json);
+                    writer.write(batch).unwrap();
+                    writer.finish().unwrap();
+                    buf += &String::from_utf8_lossy(&rb_json);
+                    buf += ",";
+                }
+                f_res.write_all(format!("[{buf}]").as_bytes()).unwrap();
```

Comment:
Can we use log? e.g.
```suggestion
                error!("case_id:{case_id}, real_output: [{buf}]");
```